### DISCUSSION TITLE
Canceling queries (AbortController / AbortSignal)

### DISCUSSION THREAD

**Discussion by sixers** [prisma/prisma#12720] [Q&A]
Does Prisma support cancelling queries? Ideally with `AbortController` / `AbortSignal`.

**Comment by austincrim** [comment]
Hey @sixers 👋 ,

We do not support query cancellation with `AbortController`, though it sounds like an interesting idea! Would mind opening a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with more details?

**Comment by sixers** [comment]
@austincrim Does Prisma support any way of cancelling queries?

Yes, I can create a feature request.

**Reply by austincrim** [reply]
We don't currently have an API for cancelling queries. Could you help me understand your use case a bit more?

**Reply by sixers** [reply]
Cancelling long running queries.
 
Example 1: Background job running some long queries is asynchronously cancelled. The db queries should also be immediately cancelled so that we can cleanup and close the job.
Example 2: We run HTTP request and db query (via prisma) in parallel. HTTP request fails, so there's no point in waiting for prisma query to finish - it can be cancelled.

**Comment by GuyDev1** [comment]
@sixers Did you open a feature request? is this issue handled or still waiting? It can really help us too

**Reply by nickdegiacomo** [reply]
looks like it's here:
https://github.com/prisma/prisma/issues/15594

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Comment by lawrencenika** [comment]
would really love to have this feature!
